packages/contracts/src/systems/SettleGameSystem.sol

```diff
@@ -37,54 +38,99 @@ contract SettleGameSystem is System {
             for (uint32 i = 0; i < all_players.length; i++) {
                 address playerAddr = all_players[i];
 
-                PlayerData memory playerData = Player.get(playerAddr);
+                (bool success, ) = payable(playerAddr).call{value: prize / all_players.length}("");
+                require(success, "SettleGameSystem: failed to transfer prize to player");
+                Player.deleteRecord(playerAddr);
+            }
+            Game.deleteRecord();
+            return true;
+        }
+        
+        
+        // require(Game.getCurrentRound() > 1, "SettleGameSystem: should not calle settleGame before round 2");
 
-                if (playerData.status == PlayerStatus.DEAD) {
-                    payable(playerAddr).transfer(address(this).balance);
-                }
+        uint256 start_time = gameData.startTime;
+        uint256 current_round = gameData.currentRound;
+        uint256 penultimate_deadline = start_time + (current_round - 2) * Config.getRoundTimeLength();
+        uint256 last_deadline = start_time + (current_round - 1) * Config.getRoundTimeLength();
+
+        console2.log("prize: ", prize);
+
+        // check the number of players
+        for (uint32 i = 0; i < all_players.length; i++) {
+            address playerAddr = all_players[i];
+
+            PlayerData memory playerData = Player.get(playerAddr);
+
+            if (playerData.lastCheckedTime > penultimate_deadline
+                    && playerData.lastCheckedTime < last_deadline) {
+                numLastAlives++;
+            }
 
-                if (playerData.lastCheckedTime > penultimate_deadline && playerData.lastCheckedTime < last_deadline) {
-                    if (numLastAlives != 0) {
-                        payable(playerAddr).transfer(prize / numLastAlives);
-                    } else {
-                        payable(playerAddr).transfer(prize / all_players.length);
-                    }
+            console2.log("player: ", playerAddr);
+            console2.log("player status: ", playerData.lastCheckedTime);
+        }
+
+        console2.log("Current round: ", current_round);
+        console2.log("penultimate_deadline: ", penultimate_deadline);
+        console2.log("last_deadline: ", last_deadline);
+        console2.log("block time", block.timestamp);
+
+        console2.log("numAlives: ", numAlives);
+        console2.log("numLastAlives: ", numLastAlives);
+
+        // if there is no player,
+        // distribute the prize to the players who were alive in last round
+        for (uint32 i = 0; i < all_players.length; i++) {
+            address playerAddr = all_players[i];
+
+            if (Player.getLastCheckedTime(playerAddr) > penultimate_deadline
+                    && Player.getLastCheckedTime(playerAddr) < last_deadline) {
+                if (numLastAlives != 0) {
+                    (bool success, ) = payable(playerAddr).call{value: prize / numLastAlives}("");
```

Comment:
we should check whether the success is true. Or we can directly use opnezepplin's `Address.sendValue`

Reply:
https://github.com/StayWarmLabs/StayWarm/blob/dev/syora/packages/contracts/src/systems/SettleGameSystem.sol#L122

Does this line work for checking?